frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -73,16 +73,17 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
       path: `/spec/storageDeviceSets/0/count`,
       value: presentCount + 1,
     };
-    props
-      .handlePromise(k8sPatch(OCSServiceModel, ocsConfig, [patch]))
-      .then(() => {
-        setProgress(false);
-        close();
-      })
-      .catch((error) => {
-        setError(error);
-        setProgress(false);
-      });
+    if (!storageClass) {
+      setError('No StorageClass selected');
+      setProgress(false);
+    } else {
+      props
+        .handlePromise(k8sPatch(OCSServiceModel, ocsConfig, [patch]))
+        .then(() => {
+          close();
+        })
+        .catch(() => null);
+    }
```

Comment:
This code looks wrong to me. handlePromise does not expose `setError` or `setProgress` to the user. There is a bigger issue here. 
/hold

Reply:
i think you missed these lines https://github.com/openshift/console/pull/5071/files#diff-ecff7031320d48d9e801ba99a2fb7188R38-R39

Reply:
You are not manually setting the progress like the code that you just deleted. Hence loading state and error will not be correctly reflected for ALL cases. Technically the deleted code is incorrect as it does not use handlePromise in the correct manner, but mixing the right way and the wrong way causes issues.

Reply:
Okay then @cloudbehl point makes sense, sorry I missed that in haste.
https://github.com/openshift/console/pull/5071/files#r409588659